src/main/surfrdMod.F90

```diff
@@ -613,6 +614,18 @@ subroutine surfrd_cftformat( ncid, begg, endg, wt_cft, cftsize, natpft_size )
        fert_cft = 0.0_r8
     end if
 
+    if ( cft_size > 0 )then
+       call ncd_io(ncid=ncid, varname='irrigation_method', flag='read', data=irrig_method, &
```

Comment:
I actually think that irrigation_method shouldn't be on the surface dataset, but a separate stream that can be handled independently, like the pop-density streams and ndep. If you do it that way, you don't have to add more complexity to mksurfdata_map, and you have better control over how it's interpolated both spatially and in time. And you keep it as a separate file that doesn't require you to make all new surface datasets, which is a task we try NOT to do. I assume this ultimately will be something that changes both in space and time correct? Reading it as a stream would both be better and easier now as well as in the future when you want it to vary in time.

Reply:
I'd ask that we defer this to a later conversation. We need to get these changes onto master in order to move forward with the isotope implementation, and I don't want to delay them for this to be rewritten to use streams (particularly because, @ekluzek I think you're the only one who is familiar enough with streams to be able to do this pretty quickly). For now we know that more work is needed to get this spatially-varying bit to work out-of-the-box (see #565 ): what's here now is just enough for further testing / development.

Reply:
I think it's OK to defer this for later. This bit right now doesn't actually add this to mksurfdata nor create new surface datasets. But, rather than do that I think we should move it to a streams file. If I'm the only familiar with it, more people should get spunup on it. People do the solutions that they are most familiar with, even if there are better options available.

Reply:
@ekluzek I'd suggest that, at a Thursday CTSM-Software meeting, you facilitate a discussion on streams vs. surface dataset / landuse.timeseries for things like this. It feels like we need a (written-down) policy about where we want things to go moving forward which accounts for various factors at play - and I'd like to hear from you as well as scientists about the arguments for and against the different options.